app/components/post_attachment_opengraph/post_attachment_opengraph.js

```diff
@@ -372,3 +336,55 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
         },
     };
 });
+
+const getBestImageUrl = (data, imagesMetadata, deviceHeight, deviceWidth, isReplyPost) => {
```

Comment:
It seems like these 5 arguments always get built from `props`? May be best to destructure all from `props` within `getBestImageUrl`. Or pass `props` as the only argument to this function, if that's needs to be done for some reason.

Reply:
@chuttam imagesMetadata and isReplyPost are not always built in with props.  I did add a check for imagesMetadata.

I'm from a camp of passing what's needed, rather than the entire object of props so you can see what are the arguments needed and use accordingly.  Especially when I factor our getBestImageUrl method out of the class onto it's own.  If this method ever move into a common utility file, passing argument of just props could be hard to read.  But then again, we have the argument 'data', which is an object.

@migbot i agree. Thought about that but still trying to adapt to Mattermost culture.